barretenberg/cpp/src/barretenberg/commitment_schemes/shplonk/shplonk.hpp

```diff
@@ -97,48 +76,67 @@ template <typename Curve> class ShplonkProver_ {
-        std::span<const Polynomial> witness_polynomials,
-        Polynomial&& batched_quotient_Q,
+    static ProverOpeningClaim<Curve> compute_partially_evaluated_batched_quotient(
+        std::span<const ProverOpeningClaim<Curve>> opening_claims,
+        Polynomial& batched_quotient_Q,
         const Fr& nu_challenge,
         const Fr& z_challenge)
     {
-        const size_t num_opening_pairs = opening_pairs.size();
+        const size_t num_opening_claims = opening_claims.size();
 
         // {ẑⱼ(r)}ⱼ , where ẑⱼ(r) = 1/zⱼ(r) = 1/(r - xⱼ)
         std::vector<Fr> inverse_vanishing_evals;
-        inverse_vanishing_evals.reserve(num_opening_pairs);
-        for (const auto& pair : opening_pairs) {
-            inverse_vanishing_evals.emplace_back(z_challenge - pair.challenge);
+        inverse_vanishing_evals.reserve(num_opening_claims);
+        for (const auto& claim : opening_claims) {
+            inverse_vanishing_evals.emplace_back(z_challenge - claim.opening_pair.challenge);
         }
         Fr::batch_invert(inverse_vanishing_evals);
 
         // G(X) = Q(X) - Q_z(X) = Q(X) - ∑ⱼ ρʲ ⋅ ( fⱼ(X) − vⱼ) / ( r − xⱼ ),
         // s.t. G(r) = 0
-        Polynomial G(std::move(batched_quotient_Q)); // G(X) = Q(X)
+        Polynomial G(batched_quotient_Q); // G(X) = Q(X)
 
         // G₀ = ∑ⱼ ρʲ ⋅ vⱼ / ( r − xⱼ )
         Fr current_nu = Fr::one();
         Polynomial tmp(G.size());
-        for (size_t j = 0; j < num_opening_pairs; ++j) {
-            // (Cⱼ, xⱼ, vⱼ)
-            const auto& [challenge, evaluation] = opening_pairs[j];
-
+        size_t idx = 0;
+        for (const auto& claim : opening_claims) {
             // tmp = ρʲ ⋅ ( fⱼ(X) − vⱼ) / ( r − xⱼ )
-            tmp = witness_polynomials[j];
-            tmp[0] -= evaluation;
-            Fr scaling_factor = current_nu * inverse_vanishing_evals[j]; // = ρʲ / ( r − xⱼ )
+            tmp = claim.polynomial;
+            tmp[0] -= claim.opening_pair.evaluation;
+            Fr scaling_factor = current_nu * inverse_vanishing_evals[idx]; // = ρʲ / ( r − xⱼ )
 
             // G -= ρʲ ⋅ ( fⱼ(X) − vⱼ) / ( r − xⱼ )
             G.add_scaled(tmp, -scaling_factor);
 
             current_nu *= nu_challenge;
+            idx++;
         }
 
         // Return opening pair (z, 0) and polynomial G(X) = Q(X) - Q_z(X)
-        return { .opening_pair = { .challenge = z_challenge, .evaluation = Fr::zero() }, .witness = std::move(G) };
+        return { .polynomial = G, .opening_pair = { .challenge = z_challenge, .evaluation = Fr::zero() } };
     };
+
+    /**
+     * @brief Returns a batched opening claim equivalent to a set of opening claims consisting of polynomials, each
+     * opened at a single point.
+     *
+     * @param commitment_key
+     * @param opening_claims
+     * @param transcript
+     * @return ProverOpeningClaim<Curve>
+     */
+    static ProverOpeningClaim<Curve> prove(const std::shared_ptr<CommitmentKey<Curve>>& commitment_key,
```

Comment:
I think consolidating the work into this method was the right thing but to me this further highlights the inadequacy of the names prove/verify here and in many other places. I think what I would actually prefer is to use `execute()` virtually everywhere. This PR may not be the place to do it though

Reply:
will add an issue